doc/src/manual/arrays.md

```diff
@@ -793,38 +793,46 @@ Indexing by a boolean vector `B` is effectively the same as indexing by the
 vector of integers that is returned by [`findall(B)`](@ref). Similarly, indexing
 by a `N`-dimensional boolean array is effectively the same as indexing by the
 vector of `CartesianIndex{N}`s where its values are `true`. A logical index
-must be a vector of the same length as the dimension it indexes into, or it
-must be the only index provided and match the size and dimensionality of the
-array it indexes into. It is generally more efficient to use boolean arrays as
-indices directly instead of first calling [`findall`](@ref).
+must be a array of the same shape as the dimension(s) it indexes into, or it 
+must be the only index provided and match the shape of the one-dimensional 
+reshaped view of the array it indexes into. It is generally more efficient 
```

Comment:
This is still not entirely accurate as `CartesianIndex{0}` could always be ignored:
```julia
julia> a = randn(2,2); I = rand(Bool, 4);

julia> a[CartesianIndex(),I,CartesianIndex()] == a[I] # Both on LTS and master.
true
```
But I wonder do we need to mention it here.